src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -1416,7 +1417,11 @@ codet java_bytecode_convert_methodt::convert_instructions(
     else if(statement==patternt("?ipush"))
     {
       assert(results.size()==1);
-      results[0]=typecast_exprt(arg0, java_int_type());
+      namespacet ns(symbol_table);
+      // Force a simplify here in case we have --no-simplify
+      // because the simplifier is the only pass that knows how to
+      // deal with this cast from integer_typet:
+      results[0]=simplify_expr(typecast_exprt(arg0, java_int_type()), ns);
```

Comment:
Ah, my Java knowledge is rather limited -- I had assumed what was true in the source was also true of bytecode. Well, in fact the bytecode should have the bit width sorted out already, or at least your statement mentioning `int8_t` hints at that?

Yes, converting `int8_t` to `integer` shouldn't lose information, but it yields an expression that the backend cannot deal with.